### PR TITLE
Ported rebirth. Add unit tests. Added rebirth route. Added integration tests

### PR DESCRIPTION
**Pull request by TheHollidayInn**
#6887 User ops rebirth
### Changes

Ported rebirth. 
Added unit tests. 
Add rebirth route. 
Added integration tests

---

UUID: 206039c6-24e4-4b9f-8a31-61cbb9aa3f66

### CODE REVIEW

File: common/script/ops/resetGear.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  _.each(['equipped', 'costume'], function resetUserGear (type) {
+    gear[type] = {};
+    gear[type].armor = 'armor_base_0';
+    gear[type].weapon = 'weapon_warrior_0';
+    gear[type].head = 'head_base_0';
+    gear[type].shield = 'shield_base_0';
+  });
+
+  _.each(gear.owned, function resetOwnedGear (v, k) {
```
Comment (paglias) [inline common/script/ops/resetGear.js:17]: @TheHollidayInn this is working even if it's a mongoose object?

Comment (TheHollidayInn) [inline common/script/ops/resetGear.js:17]: Yea, it iterates fine. The problem is that it iterates over hidden keys. I added a condition to check those keys, and any other foreign keys.

Comment (paglias) [inline common/script/ops/resetGear.js:17]: oh cool, can you add a comment about it?

**Comment by TheHollidayInn** [comment]
Updated!

**Pull request merged**